File: scripts/algorithms/L/Longest Continuous Increasing Subsequence/Longest Continuous Increasing Subsequence.py
# Runtime: 416 ms (Top 5.00%) | Memory: 15.4 MB (Top 49.36%)

class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        count=0
        for i in range(len(nums)):
            a=nums[i]
            c=1
            for j in range(i+1, len(nums)):
                if nums[j]>a:
                    a=nums[j]
                    c+=1
                else:
                    break
            count=max(count, c)
        return count

File: scripts/algorithms/L/Longest Continuous Increasing Subsequence/Longest Continuous Increasing Subsequence.java
// Runtime: 1 ms (Top 100.00%) | Memory: 45.4 MB (Top 76.62%)
class Solution {
    public int findLengthOfLCIS(int[] nums) {
        int count = 1;
        int maxCount = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                count++;
            } else {
                maxCount = Math.max(count, maxCount);
                count = 1;
            }
        }
        maxCount = Math.max(count, maxCount);
        return maxCount;
    }
}